File: src/types/models/displays.ts
export interface Display {
  name: string;
  resolution_horizontal: number;
  resolution_vertical: number;
  x: number;
  y: number;
  width?: number;
  height?: number;
  is_primary?: boolean;
  pixel_clock?: number;
  refresh_rate?: number;
}

export type Displays = Display[];

File: data/module/displays/displays.go
package displays

// Display represents information about a display device
type Display struct {
	ID                   string   `json:"id"`
	Name                 string   `json:"name"`
	ResolutionHorizontal int      `json:"resolution_horizontal"`
	ResolutionVertical   int      `json:"resolution_vertical"`
	X                    int      `json:"x"`
	Y                    int      `json:"y"`
	Width                *int     `json:"width"`
	Height               *int     `json:"height"`
	IsPrimary            *bool    `json:"is_primary"`
	PixelClock           *float64 `json:"pixel_clock"`
	RefreshRate          *float64 `json:"refresh_rate"`
}

// GetDisplays returns display information for the current platform
func GetDisplays() ([]Display, error) {
	return getDisplays()
}
